### ISSUE TITLE
Geppetto still not loading on first attempt

### ISSUE THREAD

**Issue by pgleeson** [OpenSourceBrain/geppetto-osb#353]
@filippomc I'm still having problems loading the 3D view in Geppetto most of the time. It worked once today when not logged in and on Firefox, but after logging in it didn't and on Chrome also, even in private windows it won't load, with no warning/error.

**Comment by pgleeson** [comment]
Again, just listing this here as it may be to do with www/v1 changes.

**Comment by filippomc** [comment]
The error itself doesn't seem related to the v1 change: it's an error on geppetto which appears as a blank page (being a blank page being a consequence of the v1 switch). I've noticed that reloading the page often solves

**Comment by pgleeson** [comment]
Still an issue. Tried detailed cell tutorial on Firefox, Chrome and Safari:

<img width="1157" alt="Screenshot 2023-11-09 at 11 40 35" src="https://github.com/OpenSourceBrain/OSB_homepage/assets/1556687/ae22b8d4-3475-4ce0-8b2d-d1da11b09d88">
Failed on first page load (with no warning), got there after multiple refreshes. @tarelli it may be down to some change in the interaction between redmine and geppetto frames, if it's in the back of your memory somewhere...

**Comment by filippomc** [comment]
Geppetto fails consistently if we go to the link without a session, like https://v1.opensourcebrain.org/org.geppetto.frontend/geppetto?load_project_from_url=https://raw.githubusercontent.com/OpenSourceBrain/ACnet2/master/osbSessions/sample/Sample_Session.json.
What's interesting is that the projects are not loaded when this happens going to  https://v1.opensourcebrain.org/org.geppetto.frontend

**Comment by sanjayankur31** [comment]
Devtools give me multiple errors:

![image](https://github.com/OpenSourceBrain/geppetto-osb/assets/102575/f9132aad-4af8-4cbd-83c1-982b164cb47e)

Not all of these seem related to geppetto failing. A few CORS policy blocks seem to be causing issues too?

**Comment by sanjayankur31** [comment]
@jrmartin : anything we can do help fix this? (It's quite high priority)

**Comment by sanjayankur31** [comment]
So, if one uses `v1.opensourcebrain.org` it at least works some of the time. For example: https://v1.opensourcebrain.org/projects/acnet2/models?explorer=https%3A%2F%2Fraw.githubusercontent.com%2FOpenSourceBrain%2FACnet2%2Fmaster%2FosbSessions%2Fsample%2FSample_Session.json


BUT, if one uses `www.v1.opensourcebrain.org` it will *never* work, because geppetto is being served from `v1.opensourcebrain.org`, not from `www.v1..` and so it's blocked by CORS policy and all that.

Eg:

https://www.v1.opensourcebrain.org/projects/alleninstituteneuroml?explorer=https%3A%2F%2Fraw.githubusercontent.com%2FOpenSourceBrain%2FAllenInstituteNeuroML%2Fmaster%2FCellTypesDatabase%2Fmodels%2FNeuroML2%2FSomeCells2.net.nml

@filippomc : `www.opensourcebrain.org/project/` is being re-directed to  `www.v1.open.` but maybe it needs to be redirected to `v1.open..` instead? (This is a separate issue from geppetto falling over)

**Comment by sanjayankur31** [comment]
Updating the redirect now makes sure that we always go to `v1.` (without the `www.`) so that has helped. Geppetto still falls over, but that's a different issue.

**Comment by jrmartin** [comment]
Hi @sanjayankur31 , thanks for that extra debugging information, I'm working on this and will keep you updated.

**Comment by filippomc** [comment]
Alternative solution on Redmine: fetch geppetto under the hood until it returns a valid response and only then load the iframe cc @sanjayankur31

**Comment by filippomc** [comment]
Fixed: the issue was related to an accumulator of login timestamps being returned with the login payload and slowing it down by a lot. @pgleeson FYI the anonymous user accumulated ~300K sessions. Fixed by deleting the entries from the db -> `DELETE FROM USER_LOGINTIMESTAMPS WHERE ID_OID=1;`

**Comment by sanjayankur31** [comment]
Loads very quickly now, I'll keep an eye on it. Does this mean we need to keep cleaning up the table from time to time?

**Comment by filippomc** [comment]
Not really much, it will probably take 20+ years to get to the same numbers at the current pace. It won't hurt though!
